client/go/outline/reporting/reporting_client.go

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright 2025 The Outline Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reporting
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"encoding/json"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/config"
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/connectivity"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+func CheckTCPConnectivity(tcp transport.StreamDialer, reportingUrl string) error {
```

Comment:
Do we need to export this function?
```suggestion
func checkTCPConnectivity(tcp transport.StreamDialer, reportingUrl string) error {
```